scripts/dev.sh

```diff
@@ -0,0 +1,322 @@
+export PATH="$DEV_SHIM_PATH":$PATH
+
+
+# Since launching the postgres container worked we can set up cleanup routines. This will catch CTRL-C
+function cleanup {
+  echo
+
+  if [ ! -z "${GRAPHQL_ENGINE_PID-}" ]; then
+    # This may already have been killed:
+    kill "$GRAPHQL_ENGINE_PID" &>/dev/null || true
+  fi
+
+  case "$MODE" in
+    test|postgres)
+      # Since scripts here are tailored to the env we've just launched:
+      rm -r "$DEV_SHIM_PATH"
+
+      echo_pretty "Removing $PG_CONTAINER_NAME and its volumes in 5 seconds!  PRESS CTRL-C TO ABORT."
+      sleep 5
+      docker stop "$PG_CONTAINER_NAME" 
+      docker rm -v "$PG_CONTAINER_NAME"
+    ;;
+    graphql-engine)
+    ;;
+  esac
+
+  echo_pretty "Done"
+}
+trap cleanup EXIT 
+
+wait_docker_postgres
+
+
+DB_URL="postgres://postgres:$PGPASSWORD@127.0.0.1:$PG_PORT/postgres"
+
+if [ "$MODE" = "postgres" ]; then
+  echo_pretty "Postgres logs will start to show up in realtime here. Press CTRL-C to exit and "
+  echo_pretty "shutdown this container."
+  echo_pretty ""
+  echo_pretty "You can use the following to connect to the running instance:"
+  echo_pretty "    $ $DOCKER_PSQL"
+  echo_pretty "        or..."
+  echo_pretty "    $ PGPASSWORD="$PGPASSWORD" psql -h 127.0.0.1 -p "$PG_PORT" postgres -U postgres"
+  echo_pretty ""
+  echo_pretty "Here is the database URL:"
+  echo_pretty "    $DB_URL"
+  echo_pretty ""
+  echo_pretty "If you want to launch a 'graphql-engine' that works with this database:"
+  echo_pretty "    $ $0 graphql-engine"
+  # Runs continuously until CTRL-C, jumping to cleanup() above:
+  docker logs -f --tail=0 "$PG_CONTAINER_NAME"
+
+elif [ "$MODE" = "test" ]; then 
+  #################################
+  ###     Integration tests     ###
+  #################################
+  cd "$PROJECT_ROOT/server"
+
+  export EVENT_WEBHOOK_HEADER="MyEnvValue"
+  export WEBHOOK_FROM_ENV="http://127.0.0.1:5592"
+
+  echo_pretty "Rebuilding for code coverage"
+  stack build --fast --flag graphql-engine:developer --ghc-options=-j2 --coverage 
+
+  echo_pretty "Starting graphql-engine"
+  GRAPHQL_ENGINE_TEST_LOG=/tmp/hasura-dev-test-engine.log
+  export HASURA_GRAPHQL_SERVER_PORT=8088 
+  # stopped in cleanup()
+  stack exec graphql-engine -- --database-url="$DB_URL" serve --enable-console --stringify-numeric-types \
+    --console-assets-dir ../console/static/dist  &>  $GRAPHQL_ENGINE_TEST_LOG  & GRAPHQL_ENGINE_PID=$!
+  echo -n "Waiting for graphql-engine"
+  until curl -s "http://127.0.0.1:$HASURA_GRAPHQL_SERVER_PORT/v1/query" &>/dev/null; do
+    echo -n '.' && sleep 0.2
+  done
+  echo " Ok"
+
+  cd "$PROJECT_ROOT/server/tests-py"
+  command -v pytest >/dev/null
+  if [ $? -eq 0 ]; then
+      echo_pretty "pytest is installed, so assume we don't need to run 'pip3 install -r tests-py/requirements.txt'"
+      echo_pretty "you may need to run that yourself if things don't work, or improve this script"
+  else
+      pip3 install -r requirements.txt
```

Comment:
Consider having the script set up a Python venv instead of installing dependencies globally. It’s easy to do: you basically just run `python3 -m venv <venv dir>`, then run `source <venv dir>/bin/activate.sh`. After that, you can run `pip3 install` and `pytest` as normal, and it will just work.

On my local setup, with `server/` as my working directory, I ran `python3 -m venv .python-venv`, and I run the tests by running `(source .python-venv/bin/activate.sh && cd tests-py && pytest <pytest args>)` (in a subshell to keep my shell’s environment clean). I have `.python-venv` in my global `.gitignore`, but we could also add it to this repo’s `.gitignore`.

Reply:
This is a great idea, but I think I'm going to leave this as a TODO unless you feel strongly.

Reply:
Nevermind :)